tests/testthat/test-template.R

```diff
@@ -29,21 +29,29 @@ test_that("Code blocks are evaluated and rendered correctly", {
 })
 
 test_that("UTF-8 characters in templates", {
-  template <- htmlTemplate("template-document.html", x = "")
-  html <- renderDocument(template)
-
-  # Create the string 'Δ★😎', making sure it's UTF-8 encoded on all platforms.
-  # These characters are 2, 3, and 4 bytes long, respectively.
-  pat <- rawToChar(as.raw(c(0xce, 0x94, 0xe2, 0x98, 0x85, 0xf0, 0x9f, 0x98, 0x8e)))
-  Encoding(pat) <- "UTF-8"
-  expect_true(grepl(pat, html))
-
-  # If template is passed text_ argument, make sure it's converted from native
-  # to UTF-8.
-  latin1_str <- rawToChar(as.raw(0xFF))
-  Encoding(latin1_str) <- "latin1"
-  text <- as.character(htmlTemplate(text_ = latin1_str))
-  expect_identical(charToRaw(text), as.raw(c(0xc3, 0xbf)))
+  test_template <- function(locale=""){
+    Sys.setlocale("LC_ALL", locale)
+    template <- htmlTemplate("template-document.html", x = "")
+    html <- renderDocument(template)
+
+    # Create the string 'Δ★😎', making sure it's UTF-8 encoded on all platforms.
+    # These characters are 2, 3, and 4 bytes long, respectively.
+    pat <- rawToChar(as.raw(c(0xce, 0x94, 0xe2, 0x98, 0x85, 0xf0, 0x9f, 0x98, 0x8e)))
+    Encoding(pat) <- "UTF-8"
+    expect_true(grepl(pat, html))
+
+    # If template is passed text_ argument, make sure it's converted from native
+    # to UTF-8.
+    latin1_str <- rawToChar(as.raw(0xFF))
+    Encoding(latin1_str) <- "latin1"
+    text <- as.character(htmlTemplate(text_ = latin1_str))
+    expect_identical(charToRaw(text), as.raw(c(0xc3, 0xbf)))
+
+    # Restore locale
+    Sys.setlocale("LC_ALL", "")
+  }
+  test_template("") # The default locale
+  test_template("Chinese") # Chinese locale
```

Comment:
Does this work on non-Windows platforms?

Reply:
It passes on my local machine -- but with a warning about Chinese not being available. So it's not a meaningful test.

Reply:
Maybe when you set the locale, do something like:

```R
set_locale_failed <- FALSE
tryCatch(
  Sys.setlocale(x),
  warning = function(e) set_locale_failed <<- TRUE
)

# Early return from test_template if we couldn't set the locale
if (set_locale_failed)
  return()

# do actual testing here
```

so that it it cleanly handles the warning and doesn't run irrelevant tests.